### DISCUSSION TITLE
[CHAPTER 3] - 람다표현식 part2 질문

### DISCUSSION THREAD

**Discussion by BvrPark** [Study-2-Modern-Java-In-Action/Study-2-Modern-Java-in-Action#39] [CHAPTER 3 람다 표현식 - Q&A] [answered]
### 질문 1
책에서 p.119를 보면
```java
// 생성자 참조
Supplier<Apple> c1 = Apple::new;   // -> 1번
Apple a1 = c1.get();  // -> 2번
```
이 코드에서 2번 부분에서 Apple 객체가 만들어진다고 되어있습니다.

그리고 위의 코드를 람다로 바꾸면
```java
// 람다
Supplier<Apple> c1 = () -> new Apple(); // -> 1번
Apple a1 = c1.get(); // -> 2번
```
이렇게 되었고 
저는 처음에 람다 코드 1번에서 생성자를 선언하면서 Apple객체를 만드는 것이라고 생각했습니다.

```java
//기본 코드
class AppleSupplier implements Supplier<Apple> {
    @Override
    public Apple get() {
        return new Apple();
    }
}

Supplier<Apple> c1 = new AppleSupplier();
Apple a1 = c1.get();

```
하지만 또 이렇게 바꿔보니 Apple객체가 왜 생성자 참조나 람다식의 2번 코드에서 생성되는 것인지는 어느 정도 이해가 갑니다. 

그러면
```java
// 생성자 참조
Supplier<Apple> c1 = Apple::new;

// 람다
Supplier<Apple> c1 = () -> new Apple();
```
위의 코드들은 기본 코드에서 오버라이드 부분과 `Supplier<Apple> c1 = new AppleSupplier();`부분까지를 나타내는 것이라고 생각하면 되는걸까요?
아니면 제가 잘못이해한 것인가요?

**Comment by Youngju-Jang** [comment]
```java
//기본 코드
class AppleSupplier implements Supplier<Apple> {
    @Override
    public Apple get() {
        return new Apple();
    }
}

Supplier<Apple> c1 = new AppleSupplier();
```

- 저도 말씀하신대로 여기까지 나타내는게 맞다고 생각해요. 근데 이제 따로 클래스생성 안하고 바로 구현하는거니까 `AppleSupplier` 이런이름은 따로 없는거고 

```java
// 생성자 참조
Supplier<Apple> c1 = Apple::new;

// 람다
Supplier<Apple> c1 = () -> new Apple();
```
- 이 부분은 override 부분처럼 메소드 구현부분만 작성해둔 상태고,  
- `c1.get();` 여기가 구현해둔 메소드 실행시키는 부분 ?

**Comment by JoisFe** [comment, marked as answer]
Supplier<Apple> c1 = Apple::new; 는 Supplier의 추상 메서드를 정의한 것 뿐이구요
get을 써야 new Apple()이 되는 것으로 이해하면될 것 같습니다.
